### PR TITLE
feat: allow disabling peripherals

### PR DESCRIPTION
**Pull request by alvancamp**
This PR introduces the ability to disable SuperConductor's connection to and control of peripheral devices. At a basic level there are two new settings introduced:

1. Auto-connect to all detected peripherals. This is the current behavior.
2. If auto-connect is off, peripherals can be connected to on an individual basis.

A "Panels" section has been added to the Bridges settings:
image

A "+" button has been added to the top header which shows available (but disabled) peripherals. Clicking on an item in the list will make SuperConductor try to connect to it:
image

A "Disconnect" button has been added to the connected peripheral popover from the top header. This button has two states depending on the value of the global auto-connect setting:
image
image

Closes https://github.com/SuperFlyTV/SuperConductor/issues/65

### CODE REVIEW

**Comment by alvancamp** [comment]
I'm open to bikeshedding on how various things should be named here. I'm not convinced that I've chosen the best names for things like `AvailablePeripherals` and `clientSidePeripheralSettings`.

**Review by nytamin — CHANGES_REQUESTED** [review]
I'm concerned about the implementation in `PeripheralStreamDeck.Watch` and `PeripheralXkeys.Watch`.

* The OnDevice hack seems like a bad one to me.
* There is duplicated logic between the Watch-function and `AddAvailableDevice`.
* All the recently-added static properties and methods on the `Peripheral` class makes it confusing what's happening and what the responsibilities are.

**My proposal**

I think it would be better to refactor the watch-logic completely and move _all_ of the static methods in `Peripheral` into another class (`PeripheralWatcher`?) and to separate the logic of **discovering devices** and **connecting to them**.

I would implement it something like this:

* On init, `PeripheralsHandler` sets up a single instance of `PeripheralWatcher`
  * On init, `PeripheralWatcher` sets up the device-specific watches (`PeripheralStreamDeck.Watch` etc).
* The `Watch` methods should **not** connect to the devices at all, just inform `PeripheralWatcher` about their existence.
* `PeripheralWatcher` then informs `PeripheralHandler` about the available devices.
* Depending on settings, `PeripheralHandler` chooses to connect to the devices.

File: apps/app/src/models/project/Bridge.ts [hunk 1]
```diff
@@ -13,10 +14,14 @@ export interface Bridge {
 		devices: {
 			[deviceId: string]: DeviceOptionsAny
 		}
+		peripherals: {
+			[peripheralId: string]: PeripheralSettingsAny
+		}
+		autoConnectToAllPeripherals: boolean
```
[the remark this answers is not in the document]
Comment (alvancamp) [reply apps/app/src/models/project/Bridge.ts:20]: Good catch!

**Comment by alvancamp** [comment]
> I'm concerned about the implementation in `PeripheralStreamDeck.Watch` and `PeripheralXkeys.Watch`...

This is all good feedback, thanks! I believe I have implemented the changes as desired, but let me know if you still see ways things could be improved.

One thing I'm not totally happy about is the near-duplication of logic in the device-specific Watch methods (`PeripheralStreamDeck.Watch` and `PeripheralXkeys.Watch`). They're just different enough that I couldn't figure out an elegant way of making them share code. I also couldn't think of a good way to move that logic up into `PeripheralWatcher`, which might be where it belongs.

**Comment by alvancamp** [comment]
I should also note that I do not have an Xkeys panel, so I am unable to test that part of this PR.

**Review by nytamin — APPROVED** [review]
Yes this looks much better!

I'm happy with the near-duplicated logic for xkeys and streamdeck. Since they are separate libraries and panel types, I think it's better to keep all of that logic separated.

**Pull request merged**